src/server/domain/consents/{ID}/generateChallenge.ts

```diff
@@ -0,0 +1,76 @@
+/*****
+ License
+ --------------
+ Copyright © 2017 Bill & Melinda Gates Foundation
+ The Mojaloop files are made available by the Bill & Melinda Gates Foundation under the Apache License, Version 2.0 (the 'License') and you may not use these files except in compliance with the License. You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, the Mojaloop files are distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ Contributors
+ --------------
+ This is the official list of the Mojaloop project contributors for this file.
+ Names of the original copyright holders (individuals or organizations)
+ should be listed with a '*' in the first column. People who have
+ contributed from an organization can be listed under the organization
+ that actually holds the copyright for their contributions (see the
+ Gates Foundation organization for an example). Those individuals should have
+ their names indented and be marked with a '-'. Email address can be added
+ optionally within square brackets <email>.
+ * Gates Foundation
+ - Name Surname <name.surname@gatesfoundation.com>
+
+ - Abhimanyu Kapur <abhi.kapur09@gmail.com>
+
+ --------------
+ ******/
+
+import { Request } from '@hapi/hapi'
+import { consentDB } from '../../../../lib/db'
+import { Consent } from '../../../../model/consent'
+import { putConsentId } from '../../../../shared/requests'
+import { promisify } from 'util'
+import { randomBytes } from 'crypto'
+const Enum = require('@mojaloop/central-services-shared').Enum
+
+export const isConsentRequestValid = function (request: Request, consent: Consent): boolean {
+  const fspiopSource = request.headers[Enum.Http.Headers.FSPIOP.SOURCE]
+
+  return (consent && consent.initiatorId === fspiopSource)
+}
+
+export const genChallenge = async function (request: Request, consent: Consent): Promise<void> {
```

Comment:
I suggest renaming this function to `generateChallenge`. You should skip declaring a const variable with anonymous function as a value (it is antipattern). Instead, export named function.

Reply:
please add documentation for this function -> the description where you explain what for it should be used, what logical steps are done (generation of the challenge value, store it, update the client about it via async PUT

Reply:
This is being a bit nitpicky here but the function is named `generateChallenge` but what you do is you generate the challenge, followed by storing it, following by updating the client via a PUT. Thus the name `generateChallenge` is not as apt as it should be.

Additionally, I feel like we should separate the concerns apart. In `generateChallenge` should be code to solely generate a challenge, whereas both the outgoing `PUT` request as well as the storing into the db should be abstracted into two other functions and called accordingly.

Reply:
Agree with the thoughts on function naming and modularization. the putConsentID is actually a separate function already which is building the body and then making the PUT call. Will refactor the rest. Thought I think its useful to leave the main function name as generateChallenge to signify it as the main function with the domain  file?